packages/app/src/components/admin/AdminPage.tsx

```diff
@@ -0,0 +1,87 @@
+import React, { useCallback, useContext, useEffect } from 'react';
+import {
+  Content,
+  ErrorPage,
+  Header,
+  HeaderTabs,
+  Page,
+} from '@backstage/core-components';
+import DynamicRootContext from '../DynamicRoot/DynamicRootContext';
+import Grid from '@mui/material/Grid';
+import Box from '@mui/material/Box';
+import { Link, useLocation, useNavigate } from 'react-router-dom';
+
+const tabs = [
+  {
+    id: 'rbac',
+    label: 'RBAC',
+    tabProps: { to: '/admin/rbac', component: Link },
+  },
+  {
+    id: 'plugins',
+    label: 'Plugins',
+    tabProps: { to: `/admin/plugins`, component: Link },
+  },
+];
+
+export const AdminPage = () => {
+  const { mountPoints } = useContext(DynamicRootContext);
+  const navigate = useNavigate();
+  const { pathname } = useLocation();
+  const [parent, selectedTab] = pathname.slice(1, pathname.length).split('/');
+  const availableTabs = tabs.filter(({ id }) => {
+    return typeof mountPoints[`admin.page.${id}/cards`] !== 'undefined';
+  });
+  const invalidTab = useCallback(
+    (name: string) =>
+      name === undefined ||
+      availableTabs.find(tab => tab.id === name) === undefined,
+    [availableTabs],
+  );
+  const { id: defaultTabId } = availableTabs[0];
+  // deal with any routing outliers by navigating to the first available tab
+  useEffect(() => {
+    if (invalidTab(selectedTab)) {
+      navigate(`/${parent}/${defaultTabId}`, { replace: true });
+    }
+  }, [defaultTabId, invalidTab, navigate, parent, selectedTab]);
+  // this page shouldn't be routed to if there's no mount points for
+  // it, but there's edge cases where it can happen
+  if (availableTabs.length === 0) {
+    return (
+      <ErrorPage
+        status="no content available"
+        statusMessage="No admin mount points are configured"
+      />
+    );
+  }
+  if (invalidTab(selectedTab)) {
+    // in this case the page will navigate to a tab, so avoid
```

Comment:
which tab will it navigate to? in this case? and if am getting it right currently on rbac and plugins tab would be allowed right? in future, if any other tab needs to be added then it will require code change in this file?

Reply:
Correct.  So this page will either navigate to the "rbac" tab or the "plugin" tab depending on what's available, and if there's no configured tab, for example let's say the configuration for all of this is removed while a user is at one of these pages then the page resorts to showing the standard error page which provides the user a link out of this area.

Reply:
Oh yes, and for this page additional tabs can be added by modifying [this array](https://github.com/janus-idp/backstage-showcase/pull/932/files#diff-1913f62d77a8135d4b4a3c8abd92927824cf8fbd41f5e94b098880092229257cR14-R25)